### PR TITLE
Create gosec.yml

### PR DESCRIPTION
**Pull request by faddat**
Informal systems gosec tool.

### CODE REVIEW

File: .github/workflows/gosec.yml [hunk 1]
```diff
@@ -0,0 +1,20 @@
+name: Run Gosec
+on:
+  push:
+    branches:
+      - master
```
Comment (giansalex) [inline .github/workflows/gosec.yml:5]: Juno uses `main` branch

File: .github/workflows/gosec.yml [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+jobs:
+  tests:
+    runs-on: ubuntu-latest
+    env:
+      GO111MODULE: on
+    steps:
+      - name: Checkout Source
+        uses: actions/checkout@v2
+      - name: Run Gosec Security Scanner
+        uses: securego/gosec@master
```
Comment (giansalex) [inline .github/workflows/gosec.yml:18]: but this uses original `gosec`, maybe we need to use `informalsystems/gosec@master` .

Comment (faddat) [reply .github/workflows/gosec.yml:18]: whoa thank you.  I need to change this, it is from the informal gosec page, a copy/paste.  

<3

Comment (giansalex) [reply .github/workflows/gosec.yml:18]: yes, they need to update it.

File: .github/workflows/gosec.yml (hunk 1 above)
Comment (JakeHartnell) [inline .github/workflows/gosec.yml:5]:
```suggestion
      - main
```

File: .github/workflows/gosec.yml [hunk 3]
```diff
@@ -0,0 +1,20 @@
+name: Run Gosec
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+    branches:
+      - master
```
Comment (giansalex) [inline .github/workflows/gosec.yml:8]:
```suggestion
      - main
```

**Comment by the-frey** [comment]
This is still in progress right?

**Review by giansalex — APPROVED** [review]

**Review by the-frey — APPROVED** [review]
LGTM

**Pull request merged**